compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -143,6 +145,11 @@ object Contexts {
     protected def gadt_=(gadt: GADTMap): Unit = _gadt = gadt
     final def gadt: GADTMap = _gadt
 
+    /** The terms currently known to be non-null (in spite of their declared type) */
+    private[this] var _flowFacts: FlowFacts = _
```

Comment:
Adding a new field to `Context` should only be done after careful consideration, since `Context` is together with `::` the class that allocates most memory in the compiler, and excessive allocation ruins performance. 

There are three classes of properties in Contexts: 

 - Fields
 - Entries in the store
 - Entries in the `Property` map.

Here's the criteria to choose between them:

 - Field should only be used if both reads and functional updates are very common, (e.g. used in all compiler phases).
 - Store entries can be used if reads are common but updates are infrequent.
 - Properties can be used if both reads and updates are infrequent. 

I am not sure what the best category for `flowFacts` is. My guess would be store entries, as comparisons with null are quite rare.

Reply:
Done.